crates/full-node/sov-sequencer/src/preferred/block_executor.rs

```diff
@@ -596,9 +616,12 @@ impl<S: Spec, Rt: Runtime<S>> RollupBlockExecutor<S, Rt> {
                 next_visible_rollup_height,
                 HexString(next_visible_root.namespace_root(sov_state::ProvableNamespace::User))
             );
-            self.state_roots
-                .insert(next_visible_rollup_height, next_visible_root);
+            state_roots.insert(next_visible_rollup_height, next_visible_root);
         }
+        state_roots
+    }
+
+    fn update_kernel_with_user_state_root(&mut self) {
         // take all roots greater than self.started_from
         for (height, root) in self.state_roots.iter() {
```

Comment:
@preston-evans98 I kept the behavior the same as in the original code, but should we be looping over all self.state_roots or only the new entries returned by `get_state_roots`? Will update the comment in 625.